File: src/main/scala/cn/morningwn/t0501_t1000/t0901_t1000/T0944.scala
package cn.morningwn.t0501_t1000.t0901_t1000

import org.junit.Test

import scala.util.control._

/**
 * <a href="https://leetcode.cn/problems/delete-columns-to-make-sorted/">944. 删列造序</a>
 *
 * @author moringwn
 * @create in 2022/5/12 10:49
 */
class T0944 {

    @Test
    def test(): Unit = {
        val strs: Array[String] = Array("cba", "daf", "ghi")

        println(minDeletionSize(strs))
    }

    def minDeletionSize(strs: Array[String]): Int = {
        val length: Int = strs(0).length
        val n: Int = strs.length
        var count: Int = 0
        val loop = new Breaks

        for (i <- 0 until length) {
            var pre: Char = strs(0).charAt(i);
            loop.breakable {
                for (j <- 1 until n) {
                    val cur: Char = strs(j).charAt(i)
                    if (cur < pre) {
                        count += 1
                        loop.break
                    } else {
                        pre = cur
                    }
                }
            }

        }

        count
    }

}

File: src/main/java/com/github/morningwn/t0501_t1000/t0901_t1000/t0941_t0950/T0944.java
package com.github.morningwn.t0501_t1000.t0901_t1000.t0941_t0950;

import com.github.morningwn.prefab.Out;
import org.junit.Test;

/**
 * <a href="https://leetcode.cn/problems/delete-columns-to-make-sorted/">944. 删列造序</a>
 *
 * @author moringwn
 * @create in 2022/5/12 10:36
 */
public class T0944 {

    @Test
    public void test() {
        String[] strs = {"cba"};

        Out.println(minDeletionSize(strs));
    }

    public int minDeletionSize(String[] strs) {
        int length = strs[0].length();
        int n = strs.length;
        int count = 0;

        for (int i = 0; i < length; i++) {
            char cur = strs[0].charAt(i);
            for (int j = 1; j < n; j++) {
                char tmp = strs[j].charAt(i);
                if (tmp < cur) {
                    count++;
                    break;
                } else {
                    cur = tmp;
                }
            }
        }

        return count;
    }
}
